scripts/build.sh

```diff
@@ -0,0 +1,293 @@
+}
+
+stress() {
+	docker run --rm -v $(pwd):$BUILDMNT -w $BUILDMNT $BUILD_IMAGE /bin/sh -c \
+    "GODEBUG=x509ignoreCN=0 go test ${VERBOSE:+-v} -timeout 60s -coverprofile=coverage.txt -covermode=atomic $GOTARGET/test/stress/..."
+}
+
+integration() {
+	docker run --rm \
+        -v $(pwd):$BUILDMNT \
+        -v /tmp/artifacts:/tmp/artifacts \
+        -v ${HOME}/.kube/config:/root/.kube/kubeconfig \
+        --env KUBECONFIG=/root/.kube/kubeconfig \
+        -w $BUILDMNT \
+        --env ARTIFACTS_DIR=/tmp/artifacts \
+        --env SONOBUOY_CLI=$SONOBUOY_CLI \
+        --env GODEBUG=x509ignoreCN=0 \
+        --network host \
+        $BUILD_IMAGE \
+    go test ${VERBOSE:+-v} -timeout 3m -tags=integration $GOTARGET/test/integration/...
+}
+
+lint() {
+	docker run --rm -v $(pwd):$BUILDMNT -w $BUILDMNT $BUILD_IMAGE /bin/sh -c \
+    "golint -set_exit_status ${VERBOSE:+-v} -timeout 60s $GOTARGET/cmd/... $GOTARGET/pkg/..."
+}
+
+vet() {
+	docker run --rm -v $(pwd):$BUILDMNT -w $BUILDMNT $BUILD_IMAGE /bin/sh -c \
+    "CGO_ENABLED=0 go vet ${VERBOSE:+-v} -timeout 60s $GOTARGET/cmd/... $GOTARGET/pkg/..."
+}
+
+pre() {
+	wget https://github.com/estesp/manifest-tool/releases/download/v1.0.1/manifest-tool-linux-amd64 \
+        -O manifest-tool
+	chmod +x ./manifest-tool
+	echo $DOCKERHUB_TOKEN | docker login --username sonobuoybot --password-stdin
+}
+
+# TODO: Make it easy to build single container for a specific arch
+build_container_dockerfile_image() {
+	docker build \
+        -t $REGISTRY/$2:$IMAGE_VERSION \
+        -t $REGISTRY/$2:$IMAGE_TAG \
+        -t $REGISTRY/$2:$IMAGE_BRANCH \
+        -t $REGISTRY/$2:$GIT_REF_SHORT \
+        -f $1 \
+		.
+}
+
+linux_containers() {
+	for arch in $LINUX_ARCH; do
+		if [ "$arch"="amd64" ]; then
+			sed -e "s|BASEIMAGE|$AMD_IMAGE|g" \
+			    -e 's|CMD1||g' \
+			    -e 's|BINARY|build/linux/amd64/sonobuoy|g' Dockerfile > Dockerfile-$arch
+			build_container_dockerfile_image Dockerfile-$arch sonobuoy
+			build_container_dockerfile_image Dockerfile-$arch sonobuoy-$arch
+	    elif [ "$arch"="arm64" ]; then
+			sed -e "s|BASEIMAGE|$ARM_IMAGE|g" \
+			    -e 's|CMD1||g' \
+			    -e 's|BINARY|build/linux/arm64/sonobuoy|g' Dockerfile > Dockerfile-$arch
+			build_container_dockerfile_image Dockerfile-$arch sonobuoy-$arch
+		else
+			echo "Linux ARCH unknown"
+        fi
+	done
+}
+
+windows_containers() {
+	for arch in $WIN_ARCH; do
+		if [ "$arch"="amd64" ]; then
+			sed -e "s|BASEIMAGE|$WIN_IMAGE|g" \
+			    -e 's|BINARY|build/windows/amd64/sonobuoy.exe|g' DockerfileWindows > DockerfileWindows-$arch
+			build_container_dockerfile_image DockerfileWindows-$arch sonobuoy
+			build_container_dockerfile_image DockerfileWindows-$arch sonobuoy-win-$arch
+		else 
+			echo "Windows ARCH unknown"
+        fi
+	done
+}
+
+build_binary_GOOS_GOARCH() {
+	LDFLAGS="-s -w -X $GOTARGET/pkg/buildinfo.Version=$GIT_VERSION -X $GOTARGET/pkg/buildinfo.GitSHA=$GIT_REF_LONG"
```

Comment:
Should add a comment here. The LDFLAGS is the only real issue with the script. It was a bit of a pain to work out because our env vars are making up the string which is being saved into a variable which is being passed as an argument into the container and then (previously) was being interpreted by that shell.

As a comment below says; I removed one layer of shell and then worked it out.

Very good/helpful article here: http://mywiki.wooledge.org/BashFAQ/050